vxquery-core/src/main/java/org/apache/vxquery/metadata/VXQueryCollectionOperatorDescriptor.java

```diff
@@ -129,11 +137,26 @@ public void nextFrame(ByteBuffer buffer) throws HyracksDataException {
                                 Iterator<File> it = FileUtils.iterateFiles(collectionDirectory,
                                         new VXQueryIOFileFilter(), TrueFileFilter.INSTANCE);
                                 while (it.hasNext()) {
-                                    File xmlDocument = it.next();
-                                    if (LOGGER.isLoggable(Level.FINE)) {
-                                        LOGGER.fine("Starting to read XML document: " + xmlDocument.getAbsolutePath());
+                                    File file = it.next();
+                                    if (file.getName().toLowerCase().endsWith(".xml")) {
+                                        if (LOGGER.isLoggable(Level.FINE)) {
+                                            LOGGER.fine("Starting to read XML document: " + file.getAbsolutePath());
+                                        }
+                                        parser.parseElements(file, writer, tupleIndex);
+                                    } else if (file.getName().toLowerCase().endsWith(".json")) {
+                                        if (LOGGER.isLoggable(Level.FINE)) {
+                                            LOGGER.fine("Starting to read JSON document: " + file.getAbsolutePath());
+                                        }
+                                        try {
+                                            input = new InputStreamReader(new FileInputStream(file));
+                                            jsonAbvs.reset();
+                                            jparser.parse(input, jsonAbvs);
+                                            FrameUtils.appendFieldToWriter(writer, appender, jsonAbvs.getByteArray(),
+                                                    jsonAbvs.getStartOffset(), jsonAbvs.getLength());
+                                        } catch (Exception e) {
```

Comment:
Could we just catch the exceptions that are thrown by the code in the try block?